modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookupBuilder.java

```diff
@@ -1680,6 +1680,7 @@ public PainlessLookup build() {
             );
         }
 
+        classesToDirectSubClasses.replaceAll((key, set) -> Set.copyOf(set)); // save some memory, especially when set is empty
```

Comment:
nit: Wouldn't this be better suited inside the PainlessLookup class that already does copyOf on the Map itself?

Reply:
I liked it here better because here we know that we have mutable `HashSet`, in the constructor it's not as visible. So if we ever change anything here that makes that copy redundant, then we will be less likely to clean things up downstream ... just my reasoning, likely will never become important :)